app.html

```diff
@@ -556,6 +591,57 @@
   return "<div class=edit><span class=title>name<br><input type=text class=myinput id=name_input value='"+value+"' onchange='SaveName()'></span></div>\n";
 }
 
+
+function SaveVariation() {
+  var variation_slider = document.getElementById('variation_slider');
+  console.log("SAVE VARIATION " + variation_slider.value);
+  if (presets[current_preset_num].VARIATION != variation_slider.value) {
+    presets[current_preset_num].VARIATION = variation_slider.value;
+    Send("variation " + variation_slider.value);
+  }
+  var variation_field = document.getElementById('variation_field');
+  if ( variation_field.value != variation_slider.value){
+     variation_field.value = variation_slider.value; //update numeric value when altered by slider
+  }
+  const preset_tags = document.getElementsByClassName('preset');
+  //	SetPreset(current_preset_num) //no need, variation is always updates directly
+  UpdatePresets();  
+}
+
+
+function SetVariation() {
+  var set_variation_field = document.getElementById('variation_field');
+  console.log("SET VARIATION " + set_variation_field.value);
+  var set_variation_slider = document.getElementById('variation_slider');
+  set_variation_slider.value = set_variation_field.value;
+  SaveVariation(); // now do the actual update
+}
+
+function UpdateVariation(value){
+  value = parseInt(value, 10);
+  var set_variation_slider = document.getElementById('variation_slider');
+  var set_variation_field = document.getElementById('variation_field');
+  var tmp = parseInt(set_variation_field.value, 10);
+  tmp += value;
+  tmp = tmp <= 0 ? 32767 : tmp >= 32767 ? 0 : tmp;  //clamp between min and max slider value, rollover to 0 when too high, rollback to 32767 when negative 
```

Comment:
could just do tmp &= 32767; instead

Reply:
Also, 32767 is allowed, so we shouldn't roll over until we hit 32768

Reply:
Clever, first version didn't roll over and just clmaped to 0 and 32767,
With roll-over this is very elegant indeed.